### PR TITLE
(#1) Add basic pushing of packages to CCR

### LINKED ISSUE #1: Add automated pushing of packages github action
To allow us to not have to push packages manually, we should have some automation in place.
As a start, we need to add a GitHub action that allows us to push packages to CCR on our behalf.

### PR DESCRIPTION
**Pull request by AdmiringWorm**
This pull request adds some very basic automation to use when pushing packages to the Chocolatey Community Repository.

This is done so we don't have to push packages manually, and instead can use a common user to push packages with in the same fashion every time.

NOTE: As it is a GitHub action, it is not known if works as expected until we actually do the push. Please scrutinize as much as possible during the review.

fixes #1

https://app.clickup.com/t/20540031/ENGTASKS-1189

https://app.clickup.com/t/20540031/ENGTASKS-1186

### CODE REVIEW

File: .github/workflows/package-pusher.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      API_KEY: ${{ secrets.CHOCO_API_KEY }}
+      SOURCE_LOCATION: "https://push.chocolatey.org"
+
+    steps:
+      - name: Install Dependencies
+        run: |
+          # Placeholder for installing dependencies
+          # choco install chocolatey-core.extension -y
+        shell: powershell
+      - name:
```
Comment (gep13) [inline .github/workflows/package-pusher.yml]: Is there a reason that this doesn't have a name?

File: .github/workflows/package-pusher.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          Get-CimInstance win32_operatingsystem -Property Caption, OSArchitecture, Version | fl Caption, OSArchitecture, Version
+          $PSVersionTable
+          choco --version
+        shell: powershell
+      - uses: actions/checkout@v3.0.0
+      - name: Pack Packages
+        env:
+          PACKAGE_PUSHES: ${{ github.event.inputs.packages }}
+        run: |
+          $packages = $env:PACKAGES_PUSHES -split ' '
```
Comment (gep13) [inline .github/workflows/package-pusher.yml]:
```suggestion
          $packages = $env:PACKAGE_PUSHES -split ' '
```

Or the line above needs to be `PACKAGES_PUSHES`

**Review by gep13 — CHANGES_REQUESTED** [review]
Noticed a couple things

File: .github/workflows/package-pusher.yml (hunk 2 above)
Comment (AdmiringWorm) [reply .github/workflows/package-pusher.yml]: That is what I get when doing a partial copy from my own repository where I use a similar action.

Good catch.

File: .github/workflows/package-pusher.yml (hunk 1 above)
Comment (AdmiringWorm) [reply .github/workflows/package-pusher.yml]: I have added a name for this one now.

File: .github/workflows/package-pusher.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          Write-Host "Packing Packages:`n$packages"
+          
+          foreach ($package in $packages) {
+            Write-Host ("{0}`n{1}`n" -f ('-'*60), "PACKAGE: $package")
+            $package_dir = Get-ChildItem -Recurse | ? { $_.Name -eq "$package.nuspec" } | Select-Object -First 1 | % Directory
+
+            if (!$package_dir) { Write-Warning "Can't find package '$package'"; continue }
+            Push-Location $package_dir
+
+            if (Test-Path update.ps1 -ea 0) { ./update.ps1 }
```
Comment (TheCakeIsNaOH) [inline .github/workflows/package-pusher.yml:45]: This is fine if the package is not embedded, but what about packages that have embedded binaries? It seems like this workflow would not include them in the package. 

Well, unless the package is both outdated and has an AU script.

Comment (AdmiringWorm) [reply .github/workflows/package-pusher.yml:45]: Packages that are located in this repository will not be AU packages.

They are meant to only be extension packages that should have everything located.

The `update.ps1` will only be there if we need to do additional work for the package if needed.

Comment (TheCakeIsNaOH) [reply .github/workflows/package-pusher.yml:45]: Ah, got it.

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@AdmiringWorm thanks for getting this started! This all looks good to me!

**Pull request merged**